src/test/lib/DataTest.cpp

```diff
@@ -2960,6 +2960,116 @@ QuicTestStreamAbortConnFlowControl(
+
+    static QUIC_STATUS ClientStreamCallback(_In_ MsQuicStream*, _In_opt_ void* ClientContext, _Inout_ QUIC_STREAM_EVENT* Event) {
+        auto TestContext = (StreamReliableReset*)ClientContext;
+        if (Event->Type == QUIC_STREAM_EVENT_SHUTDOWN_COMPLETE) {
+            TestContext->ClientStreamShutdownComplete.Set();
+        }
+        return QUIC_STATUS_SUCCESS;
+    }
+
+    static QUIC_STATUS ServerStreamCallback(_In_ MsQuicStream* Stream, _In_opt_ void* ServerContext, _Inout_ QUIC_STREAM_EVENT* Event) {
+        auto TestContext = (StreamReliableReset*)ServerContext;
+        if (Event->Type == QUIC_STREAM_EVENT_RECEIVE) {
+            TestContext->ReceivedBufferSize += Event->RECEIVE.TotalBufferLength;
+        }
+        if (Event->Type == QUIC_STREAM_EVENT_PEER_SEND_SHUTDOWN) {
+            Stream->Shutdown(QUIC_STREAM_SHUTDOWN_FLAG_ABORT_SEND);
+        }
+        return QUIC_STATUS_SUCCESS;
+    }
+
+    static QUIC_STATUS ConnCallback(_In_ MsQuicConnection*, _In_opt_ void* Context, _Inout_ QUIC_CONNECTION_EVENT* Event) {
+        if (Event->Type == QUIC_CONNECTION_EVENT_PEER_STREAM_STARTED) {
+            new(std::nothrow) MsQuicStream(Event->PEER_STREAM_STARTED.Stream, CleanUpAutoDelete, ServerStreamCallback, Context);
+        }
+        return QUIC_STATUS_SUCCESS;
+    }
+};
+
+void
+QuicTestStreamReliableReset(
+    )
+{
+    /*
+
+    Client initiates stream, connects to Server.
+    Client sets ReliableOffset (send offset) to be 6.
+    Client sends data to server
+    Client immediately calls shutdown.
+    Client should finish sending data up to 6 before actually aborting.
+    Server should get data up to 6 before ACKing the shutdown.
+
+
+    Best way to do this;
+     - artificially delay receiver callback
+     - Peer Event Started
+     - Theoretically you can block receive event...
+
+     Quic Send Flag: QUIC_SEND_FLAG_DELAY_FLAG (not always happens)
+      - Here's a send, you don't have to send it yet.
+      - wait for handshake, sleep(50)
+      - start, send (with delay), (set reliable offset), shutdown (do everything after handshake completes)
+
+    */
+    MsQuicRegistration Registration(true);
+    TEST_TRUE(Registration.IsValid());
+
+    MsQuicSettings ServerSettings;
+    MsQuicSettings ClientSettings;
+    ServerSettings.SetReliableResetEnabled(true);
+    ClientSettings.SetReliableResetEnabled(true);
+    ServerSettings.SetPeerBidiStreamCount(1);
+
+    MsQuicConfiguration ServerConfiguration(Registration, "MsQuicTest", ServerSettings, ServerSelfSignedCredConfig);
+    TEST_QUIC_SUCCEEDED(ServerConfiguration.GetInitStatus());
+
+    MsQuicConfiguration ClientConfiguration(Registration, "MsQuicTest", ClientSettings, MsQuicCredentialConfig());
+    TEST_QUIC_SUCCEEDED(ClientConfiguration.GetInitStatus());
+
+    StreamReliableReset Context;
+    #define BUFFER_SIZE 10000
+    #define RELIABLE_SIZE 1000
+    uint8_t SendDataBuffer[BUFFER_SIZE];
+    uint8_t ReceiveDataBuffer[BUFFER_SIZE];
```

Comment:
You don't need `ReceiveDataBuffer` any more, right?

Reply:
Good catch! Removed.